python/lsst/ts/audiotrigger/read_serial_temp_scanner.py

```diff
@@ -89,75 +110,69 @@ class SerialTemperatureScanner(tcpip.OneClientServer):
         Is the server configured.
     first_run : bool
         Is the first run.
-    encoding : str
-        The string encoding type.
-    port : int
-        The port.
-    host : str
-        The host.
     loop : asyncio.Loop
         The current running async loop.
+    error_validator : `None`
+        The jsonschema validator for error messages.
+    fan_control_server : `FanControlServer`
+        The service that handles turning the fan on/off.
+    ess_server : `EssServer`
+        The service that handles sending data to the ESS client.
     """
 
+    data = None
+
     def __init__(
         self,
         log: logging.Logger | None = None,
-        port: int | None = 18830,
-        host: str | None = tcpip.DEFAULT_LOCALHOST,
-        encoding: str = tcpip.DEFAULT_ENCODING,
-        terminator: bytes = tcpip.DEFAULT_TERMINATOR,
-        sample_wait_time: int = 5,
-        serial=None,
-        temperature_windows: int = 8,
         simulation_mode=False,
     ):
         self.log = log
-        self.simulation_mode = simulation_mode
-
-        if not self.simulation_mode:
-            super().__init__(
-                log=self.log,
-                port=port,
-                host=host,
-                connect_callback=None,
-                monitor_connection_interval=0,
-                name="",
-                encoding=encoding,
-                terminator=terminator,
-            )
-
-        self.serial = serial
+        self.simulation_mode = bool(simulation_mode)
         self.sensor_dict = {}
-        self.sample_wait_time = sample_wait_time
+        self.sample_wait_time = 5
 
         # Fan sensor
         self.fan_sensor = ""
         self.fan_gpio = None
         self.fan_turn_on_temp = 0
         self.fan_turn_off_temp = 0
 
-        self.latest_data = {sensor_name: 0 for sensor_name in self.sensor_dict}
-        self.rolling_temperature = [0 for _ in range(temperature_windows)]
-
         if self.simulation_mode:
             self.pi = MockPio()
+            self.serial = common.device.MockDevice(
+                name="Mock Fan Control sensors",
+                device_id="/dev/ttyUSB0",
+                sensor=common.sensor.TemperatureSensor(log=self.log, num_channels=8),
+                callback_func=callback,
+                log=self.log,
+            )
         else:
             self.pi = pigpio.pi()
+            self.serial = controller.device.RpiSerialHat(
+                name="Fan Control sensors",
+                device_id="/dev/ttyUSB0",
+                sensor=common.sensor.TemperatureSensor(log=self.log, num_channels=8),
+                baud_rate=19200,
+                callback_func=callback,
+                log=self.log,
+            )
```

Comment:
Please rewrite this if/else so common parts (like the device_id) are only stored once. The only difference really is the sensor name. I think the mock sensor can handle the baud rate, can't it?

Reply:
Please create a Jira ticket to send the config and add a TODO here. I requested that during our 1n1 on Monday.

Reply:
https://github.com/lsst-ts/ts_audiotrigger/blob/8c091bc2e61feb725472fbf00de9026683f3abba/python/lsst/ts/audiotrigger/read_serial_temp_scanner.py#L175
I have the ticket and TODO here.